lib/WorseReflection/Bridge/TolerantParser/Diagnostics/DeprecatedUsageDiagnosticProvider.php

```diff
@@ -220,6 +220,36 @@ public function foo(Closure $foobar) {
                 Assert::assertEquals('Call to deprecated property "deprecated": This is deprecated', $diagnostics->at(0)->message());
             }
         );
+
+        yield new DiagnosticExample(
+            title: 'Deprecate with replacement',
+            source: <<<'PHP'
+                <?php
+
+                class SomeClass {
+                    /**
+                    * @deprecated Will be removed
+                    *
+                    * @see SomeClass::new
```

Comment:
what implies that SomeClass::new is a replacement?

Reply:
From the [standard](https://github.com/php-fig/fig-standards/blob/master/proposed/phpdoc-tags.md#54-deprecated):

> If the associated element is superseded by another it is RECOMMENDED to add a @see tag in the same 'PHPDoc' pointing to the new element.

Reply:
That's by convention.

> If [the deprecated function] is superseded by another it is RECOMMENDED to add a @see tag in the same 'PHPDoc' pointing to the new element.

https://github.com/php-fig/fig-standards/blob/master/proposed/phpdoc-tags.md#54-deprecated

Reply:
It's not really a standard.... but the "see" applies to the entire docblock, it's only coincidentally related to the deprecation message

Reply:
So it is mentioned on the official website:

https://docs.phpdoc.org/3.0/guide/references/phpdoc/tags/deprecated.html#deprecated

In the case where there's an `@deprecated` and `@see` tag on the same element, then the `@see` is understood to point to the new element to replace the deprecated one

Reply:
Seems more coincidental to me, you can add "see" to any docblock. It doesn't mean that it references the replacement. It just means "see" this other thing, which may be the replacement or something else entirely.  It's certainty added context, would be good to see what PHPStorm does.

Reply:
I had a coworker send me some screenshots of what PHPStorm does with `@deprecated` and `@see`:

Without `@see`:
![image](https://github.com/phpactor/phpactor/assets/4176520/9b79c0f0-74e1-4a40-a71a-df6b18efa106)


With `@see`:
![image](https://github.com/phpactor/phpactor/assets/4176520/91fd6399-b5b8-410e-9625-177875db8aa1)

Reply:
Yes, so deprecation or not it will show "see also"?

Reply:
Looks like it

Reply:
LSP has `relatedInformation` in the diagnostic:

https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#diagnostic

which could be used here - we can also specify `DiagnosticTag::DEPRECATED`

Reply:
in VS code it looks as you might want:

![image](https://github.com/phpactor/phpactor/assets/530801/96b9660e-7e7a-4328-b8d8-00a3316f1ee8)

in Neovim the `DiagnostcTag::DEPRECATED` causes a strike-through, but the related information is not rendered:

![image](https://github.com/phpactor/phpactor/assets/530801/6c06e77a-9b10-4882-8764-03b88e2116c0)


but maybe it could be with some tweaks, in anycase, I think `$relatedInformation` is how we should render this info, and we should also parse the class name so that we can get the location to it.

Reply:
That sounds like the much better option. Then jumping to it also would be fairly easy.

Is that only a diagnostic feature. Because it can be written in any tag. It already is part of the method documentation as part of the doc block. If we want to parse the see tag everywhere, it would be nice to have a general "is related" list of references that the see tag could provide.

Reply:
I haven't yet found the place where the WR diagnostics are converted to LSP diagnostics. Otherwise the relatedInfo thing should be easy to implement.